queries/javascript.core.scm

```diff
@@ -415,6 +404,262 @@
 ;;!! const aaa = {bbb: 0, ccc: 0};
 ;;!               **************
 (object
-  "{" @value.iteration.start.endOf
-  "}" @value.iteration.end.startOf
+  "{" @collectionKey.iteration.start.endOf @value.iteration.start.endOf
+  "}" @collectionKey.iteration.end.startOf @value.iteration.end.startOf
+)
+
+;;!! "string"
+;;!! `string`
+;;!  ^^^^^^^^
+[
+  (string)
+  (template_string)
+] @string
+
+;;!! // comment
+;;!  ^^^^^^^^^^
+(comment) @comment
+
+;;!! /\w+/
+;;!  ^^^^^
+(regex) @regularExpression
+
+[
+  (string_fragment)
+  (comment)
+  (regex_pattern)
+] @textFragment
+
+(
+  (template_string) @textFragment
+  (#child-range! @textFragment 0 -1 true true)
+)
+
+;;!! { value: 0 }
+;;!  ^^^^^^^^^^^^
+[
+  (object)
+  (object_pattern)
+] @map
+
+;;!! [ 0 ]
+;;!  ^^^^^
+[
+  (array)
+  (array_pattern)
+] @list
+
+;;!! if () {}
+;;!  ^^^^^^^^
+(if_statement) @ifStatement
+
+;;!! switch (value) {}
+;;!          ^^^^^
+;;!  -----------------
+(switch_statement
+  value: (_) @switchStatementSubject
+  (#child-range! @switchStatementSubject 0 -1 true true)
+) @_.domain
+
+;;!! foo()
+;;!  ^^^^^
+;;!! new Foo()
+;;!  ^^^^^^^^^
+[
+  (call_expression)
+  (new_expression)
+] @functionCall
+
+;;!! foo()
+;;!  ^^^
+;;!  -----
+(call_expression
+  function: (_) @functionCallee
+) @_.domain
+
+;;!! new Foo()
+;;!  ^^^^^^^
+;;!  ---------
+(new_expression
+  (arguments) @functionCallee.end.startOf
+) @functionCallee.start.startOf @_.domain
+
+;;!! class Foo {}
+;;!  ^^^^^^^^^^^^
+(
+  [
+    (class_declaration)
+    (class)
+  ] @class
+  (#not-parent-type? @class export_statement)
+)
+
+;;!! export class Foo {}
+;;!  ^^^^^^^^^^^^^^^^^^^
+(export_statement
+  [
+    (class_declaration)
+    (class)
+  ]
+) @class
+
+;;!! class Foo {}
+;;!        ^^^
+;;!  ------------
+(class_declaration
+  name: (_) @className
+) @_.domain
+(class
+  name: (_) @className
+) @_.domain
```

Comment:
I believe this breaks if you're in the `export` of an `export class ...`. Better to just add these annotations below where you define `@class`, no?

Reply:
Correct.